@codex please fix

```
> dspace@3.0.0 prebuild
> node scripts/build-processes.mjs

 ✓ tests/helmChartTls.test.ts (4 tests) 5ms
 ✓ tests/buyItemsGuardrails.test.ts (2 tests) 4ms

> dspace@3.0.0 build
> astro build

 ✓ tests/ingressTlsConfig.test.ts (11 tests) 7ms
 ✓ tests/questFormExistingQuests.test.ts (1 test) 24ms
 ✓ tests/preparePrScriptSteps.test.ts (4 tests) 6ms
 ✓ tests/postinstall.test.ts (4 tests) 10ms
18:02:19 [types] Generated 185ms
18:02:19 [build] output: "server"
18:02:19 [build] directory: /home/runner/work/dspace/dspace/frontend/dist/
18:02:19 [build] adapter: @astrojs/node
18:02:19 [build] Collecting build info...
18:02:19 [build] ✓ Completed in 290ms.
18:02:19 [build] Building server entrypoints...
Browserslist: browsers data (caniuse-lite) is 10 months old. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
 ✓ tests/packageManager.test.ts (4 tests) 7ms
 ✓ frontend/tests/questStats.test.ts (3 tests) 4ms
 ✓ tests/offlineFirstDocSync.test.ts (1 test) 10ms
 ✓ scripts/tests/fsChecks.test.ts (6 tests) 11ms
 ✓ frontend/src/components/__tests__/menuActive.spec.ts (5 tests) 4ms
 ✓ frontend/src/pages/docs/__tests__/index.spec.ts (2 tests) 150ms
 ✓ frontend/tests/docsRagComparison.test.ts (5 tests) 6ms
 ✓ scripts/tests/imageReferences.test.ts (1 test) 39ms
 ✓ tests/questPrerequisites.test.ts (1 test) 28ms
 ✓ frontend/src/pages/chat/svelte/__tests__/Integrations.spec.ts (1 test) 116ms
 ✓ tests/hardeningQuality.test.ts (2 tests) 54ms
 ✓ tests/titlesLogic.test.ts (2 tests) 7ms
 ✓ tests/roadmapChecklist.test.ts (1 test) 7ms
 ✓ tests/questRewardGrantSeparation.test.ts (1 test) 293ms
 ✓ tests/openAI.persona.test.ts (1 test) 760ms
   ✓ GPT5Chat persona integration > injects persona-specific prompt and welcome message  758ms
 ✓ tests/dCarbonOffsetDoc.test.ts (1 test) 6ms
 ✓ tests/itemValidation.test.ts (2 tests) 90ms
 ✓ tests/purgeClientStateRetry.test.ts (2 tests) 8ms
 ✓ tests/processQuality.test.ts (3 tests) 122ms
 ✓ frontend/tests/item-requirements.test.ts (3 tests) 4ms
 ✓ tests/strings.test.ts (9 tests) 7ms
 ✓ frontend/tests/questListManifest.test.ts (2 tests) 68ms
 ✓ tests/changelogContributorsGuide.test.ts (1 test) 7ms
 ✓ tests/scoreQuestOpenAI.test.ts (2 tests) 12ms
 ✓ scripts/tests/validateStagedQuests.test.ts (3 tests) 36ms
 ✓ tests/generateItemDependencies.test.ts (3 tests) 93ms
18:02:39 [build] ✓ Completed in 20.21s.

 building client (vite) 
 ✓ tests/navigateWithRetry.test.ts (2 tests) 13ms
 ✓ tests/noConflictMarkers.test.ts (1 test) 1209ms
   ✓ repository sanity ‑ no merge-conflict markers > contains no <<<<<<< or >>>>>>> markers  1207ms
 ✓ tests/k8sEnvironmentOverlays.test.ts (2 tests) 7ms
 ✓ scripts/tests/questDependencies.test.ts (2 tests) 9ms
 ✓ tests/svelteResolution.test.ts (4 tests) 518ms
 ✓ tests/architectureNotesAccessibilityBaseline.test.ts (2 tests) 4ms
 ✓ tests/questDependencyReferences.test.ts (1 test) 52ms
 ✓ tests/questIdPathConsistency.test.ts (1 test) 36ms
 ✓ tests/svelteSubpathResolver.test.ts (2 tests) 5ms
 ✓ tests/serviceWorkerConfigCache.test.ts (1 test) 7ms
 ✓ tests/noDuplicateItemIds.test.ts (1 test) 5ms
 ✓ tests/questReliabilityCoverage.test.ts (1 test) 191ms
 ✓ tests/v3ReleaseStateDoc.test.ts (3 tests) 5ms
 ✓ frontend/tests/questListClassifier.test.ts (3 tests) 7ms
 ✓ tests/changelogQuestBranchingNote.test.ts (1 test) 10ms
 ✓ tests/metricsEndpoint.test.ts (3 tests) 22ms
 ✓ tests/docsSearchFeatures.test.ts (2 tests) 7ms
 ✓ tests/questsPolishPromptSections.test.ts (3 tests) 5ms
 ✓ tests/questProcessRecoveryPaths.test.ts (1 test) 183ms
 ✓ frontend/tests/docsMarkdownSafety.test.ts (1 test) 13ms
 ✓ tests/builtinQuestSchema.test.ts (1 test) 96ms
 ✓ tests/githubTokenPersistence.test.ts (1 test) 11ms
 ✓ frontend/src/components/__tests__/QuestLinkChips.spec.ts (2 tests) 160ms
 ✓ tests/spellcheckConfig.test.ts (1 test) 6ms
 ✓ tests/pnpmFrontendFilter.test.ts (1 test) 18ms
 ✓ tests/scanSecretsScript.test.ts (2 tests) 137ms
18:02:56 [build] Rearranging server assets...
18:02:56 [build] Server built in 37.34s
18:02:56 [build] Complete!
 ✓ tests/buildOutputValidation.test.ts (1 test) 44507ms
   ✓ build output validation > should not contain /src/scripts/ paths in built files  44505ms
 ✓ tests/helmChartReleaseVersion.test.ts (1 test) 5ms
 ✓ tests/changelogDocsAICoPilots.test.ts (1 test) 6ms
 ✓ tests/customContentLoadTest.test.ts (2 tests) 1441ms
   ✓ runCustomContentLoadTest > uses default count when omitted  1407ms
 ✓ tests/compactItemListHelpers.test.ts (2 tests) 5ms
 ✓ tests/userJourneysCoverage.test.ts (1 test) 6ms
 ✓ tests/measureEcSolutionRegression.test.ts (1 test) 7ms
 ✓ tests/changelogDocsContentUpdate.test.ts (1 test) 5ms
 ✓ scripts/tests/detectZeroTests.test.ts (5 tests) 6ms
 ✓ tests/raspiSecretsDocs.test.ts (2 tests) 4ms
 ✓ tests/customQuestLoader.test.ts (1 test) 20ms
 ✓ scripts/tests/preparePrScripts.test.ts (1 test) 5ms
 ✓ tests/inject-ids.test.ts (1 test) 6ms
 ✓ tests/chartResourcesDefaults.test.ts (1 test) 29ms
 ✓ tests/cheatsAvailability.test.ts (5 tests) 4ms
 ✓ tests/helmChartGracefulShutdown.test.ts (2 tests) 6ms
 ✓ tests/changelogCloudSync.test.ts (1 test) 7ms
 ✓ tests/changelogProcessRequirementsNote.test.ts (1 test) 7ms
 ✓ tests/markdownEncodingFix.test.ts (3 tests) 5ms
 ✓ tests/docsRagDrift.test.ts (2 tests) 473ms
   ✓ docs RAG drift intent > forces the v3 release state doc for v2-only drift queries  455ms
 ✓ tests/helmChartPersistence.test.ts (2 tests) 4ms
 ✓ tests/quest-title-unique.test.ts (3 tests) 6ms
 ✓ frontend/src/components/__tests__/RemainingTime.spec.ts (2 tests) 50ms
 ✓ tests/questCopyRegression.test.ts (1 test) 5ms
 ✓ scripts/tests/validateItem.test.ts (2 tests) 8ms
 ✓ tests/chatPersonas.test.ts (1 test) 171ms
 ✓ tests/workflowPnpmCache.test.ts (3 tests) 10ms
 ✓ tests/changelogNpcChatNote.test.ts (1 test) 5ms
 ✓ frontend/tests/chatHints.test.ts (4 tests) 5ms
 ✓ tests/changelogProcessReservation.test.ts (1 test) 5ms
 ✓ tests/questTreesDoc.test.ts (3 tests) 5ms
 ✓ tests/deployEnvFluxDocs.test.ts (1 test) 4ms
 ✓ tests/networkPolicyObservability.test.ts (2 tests) 4ms
 ✓ tests/roadmapDocLinks.test.ts (1 test) 5ms
 ✓ tests/backupScript.test.ts (2 tests) 152ms
 ✓ tests/guildsDocs.test.ts (2 tests) 6ms
 ✓ scripts/tests/validateQuest.test.ts (2 tests) 34ms
 ✓ tests/seedCustomContent.test.ts (1 test) 33ms
 ✓ tests/menuToolbox.test.ts (2 tests) 6ms
 ✓ tests/roadmapPastYearsComplete.test.ts (1 test) 4ms
 ✓ tests/fix-markdown-image-paths.test.ts (1 test) 5ms
 ✓ tests/promptsCodexLinks.test.ts (1 test) 4ms
 ✓ tests/itemSelectorLocators.test.ts (1 test) 5ms
 ✓ tests/customContentItemsStore.test.ts (1 test) 16ms
 ✓ tests/helmChartProbes.test.ts (1 test) 7ms
 ✓ tests/userJourneysDoc.test.ts (2 tests) 5ms
 ✓ tests/buildWorkflowMultiArch.test.ts (2 tests) 5ms
 ✓ tests/processHardening.test.ts (1 test) 5ms
 ✓ tests/cacheVersionSync.test.ts (1 test) 6ms
 ✓ tests/dbBenchmarkCli.test.ts (1 test) 282ms
 ✓ tests/canvasDependencyAlignment.test.ts (1 test) 5ms
 ✓ tests/questSchemaValidation.test.ts (2 tests) 8ms
 ✓ tests/changelogDocsReleaseChecklist.test.ts (1 test) 5ms
 ✓ tests/playwrightConfigBrowsers.test.ts (1 test) 8ms
 ✓ frontend/src/components/__tests__/SearchBar.spec.ts (1 test) 64ms
 ✓ tests/customContentBackupImporterAccessibility.test.ts (1 test) 5ms
 ✓ tests/license.test.ts (3 tests) 6ms
 ✓ tests/questGuidelinesEditorDoc.test.ts (1 test) 4ms
 ✓ tests/ubiQuestLeaderboardCopy.test.ts (1 test) 4ms
 ✓ frontend/tests/playwrightConfigHost.test.ts (1 test) 7ms
 ✓ tests/roadmap10xQuests.test.ts (1 test) 6ms
 ✓ tests/questGraphValidation.test.ts (2 tests) 312ms
 ✓ tests/rocketsDocsContent.test.ts (1 test) 4ms
 ✓ tests/solarBasicsIcons.test.ts (5 tests) 5ms
 ✓ scripts/tests/lintStagedItemValidation.test.ts (1 test) 5ms
 ✓ scripts/tests/lintStagedQuestValidation.test.ts (1 test) 4ms
 ✓ tests/rocketsDocsPromise.test.ts (1 test) 4ms
 ✓ tests/fsChecksInvalidEncoding.test.ts (1 test) 7ms
 ✓ tests/roadmapSuborbitalLaunch.test.ts (1 test) 3ms
 ✓ tests/roadmapCustomQuests.test.ts (1 test) 4ms
 ✓ tests/changelogNotes.test.ts (1 test) 5ms
 ✓ tests/testingDocsPromises.test.ts (1 test) 5ms
 ✓ frontend/src/components/__tests__/compactItemListHelpers.spec.ts (1 test) 5ms
 ✓ tests/k3sChecklistComplete.test.ts (1 test) 5ms
 ✓ tests/docsFAQ.test.ts (1 test) 4ms
 ✓ tests/roadmapPromises.test.ts (1 test) 5ms
 ✓ tests/solarDocWind.test.ts (1 test) 6ms
 ✓ tests/docsPromptsOutages.test.ts (2 tests) 6ms
 ✓ tests/menu-active.test.ts (1 test) 10ms
 ✓ tests/sanitization.test.ts (1 test) 29ms
 ✓ tests/metricsFallback.test.ts (1 test) 42ms
 ✓ tests/itemSelectorTouch.test.ts (2 tests) 5ms
 ✓ tests/process-form-mobile.test.ts (1 test) 7ms
 ✓ tests/questCountIncrease.test.ts (1 test) 25ms
 ✓ tests/item-form-mobile.test.ts (1 test) 4ms
 ✓ tests/quest-form-mobile.test.ts (1 test) 4ms
 ✓ tests/titlesDoc.test.ts (1 test) 4ms
 ✓ tests/menu-leaderboard.test.ts (1 test) 4ms
 ✓ tests/togglePreviewId.test.ts (3 tests) 5ms
 ✓ scripts/tests/questQuality.test.ts (1 test) 6ms
 ✓ tests/npmTestCiScript.test.ts (1 test) 7ms
 ✓ tests/menu-settings.test.ts (1 test) 6ms
 ✓ tests/achievementsCatalog.test.ts (1 test) 6ms
 ✓ tests/mergePlanChecklist.test.ts (1 test) 3ms
 ✓ tests/menu-stats.test.ts (1 test) 4ms
 ✓ tests/profileTitles.test.ts (1 test) 3ms
 ✓ frontend/tests/menu.test.ts (1 test) 5ms
 ✓ tests/menuTitles.test.ts (1 test) 4ms
 ✓ frontend/tests/duration.test.ts (2 tests) 5ms
 ✓ tests/domExceptionGlobal.test.ts (1 test) 4ms
 ✓ tests/pnpmfile.test.ts (1 test) 6ms
 ✓ tests/domexception-global.test.ts (1 test) 5ms
 ✓ tests/domexception-globals.test.ts (1 test) 5ms
 ✓ tests/sysadminQuest.test.ts (1 test) 7ms
 ✓ tests/shopIndex.test.ts (1 test) 4ms
 ✓ tests/eslintPluginPresence.test.ts (1 test) 6ms
 ✓ frontend/tests/gameStateFlush.test.ts (1 test) 4ms
 ✓ tests/checkoutFetchDepth.test.ts (1 test) 10ms
 ✓ tests/npmTestRunsAllSuites.test.ts (1 test) 4ms
 ✓ scripts/tests/prettierFormatting.test.ts (1 test) 28689ms
   ✓ Prettier formatting > frontend sources are formatted  28688ms
 ✓ tests/astroConfigLogLevel.test.ts (1 test) 7ms
 ✓ tests/prismSvelte.test.ts (1 test) 11ms
 ✓ scripts/tests/dbBenchmarkScript.test.ts (1 test) 4ms
 ✓ tests/progressionBalance.test.ts (1 test) 45ms
 ✓ tests/domException.test.ts (1 test) 6ms
 ✓ tests/indexeddb.server.test.ts (1 test) 242ms
 ✓ frontend/tests/sanity.test.ts (1 test) 3ms

⎯⎯⎯⎯⎯⎯⎯ Failed Tests 4 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  frontend/tests/ensure-playwright-browsers.test.ts > ensurePlaywrightSystemDeps > invokes install-deps on linux when sentinel is missing
AssertionError: expected "spy" to be called with arguments: [ …(2) ]

Number of calls: 0

 ❯ frontend/tests/ensure-playwright-browsers.test.ts:94:35
     92|         expect(execFileSyncMock).toHaveBeenCalledTimes(1);
     93|         expect(execFileSyncMock.mock.calls[0][1]).toEqual([cliPath, 'i…
     94|         expect(writeFileSyncMock).toHaveBeenCalledWith(
       |                                   ^
     95|             depsStampPath,
     96|             expect.stringMatching(/\d{4}-\d{2}-\d{2}/)

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/4]⎯

 FAIL  frontend/tests/ensure-playwright-browsers.test.ts > ensurePlaywrightBrowsers > installs system deps before browsers when chromium is missing
AssertionError: expected "spy" to be called at least once
 ❯ frontend/tests/ensure-playwright-browsers.test.ts:285:35
    283|             'chromium-headless-shell',
    284|         ]);
    285|         expect(writeFileSyncMock).toHaveBeenCalled();
       |                                   ^
    286|     });
    287| 

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/4]⎯

 FAIL  scripts/tests/ensurePlaywrightBrowsers.test.ts > ensurePlaywrightBrowsers > installs system dependency packages with sanitized env when only placeholder proxies are present
AssertionError: expected "spy" to be called 1 times, but got 0 times
 ❯ scripts/tests/ensurePlaywrightBrowsers.test.ts:234:29
    232|         NODE_OPTIONS: '--dns-result-order=ipv4first',
    233|       });
    234|       expect(writeFileSync).toHaveBeenCalledTimes(1);
       |                             ^
    235|       expect(warnSpy).toHaveBeenCalledWith(
    236|         'Proxy environment variables point to the placeholder proxy:80…

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/4]⎯

 FAIL  scripts/tests/ensurePlaywrightBrowsers.test.ts > ensurePlaywrightBrowsers > preserves non-placeholder proxies during system deps install
AssertionError: expected "spy" to be called 1 times, but got 0 times

 ❯ scripts/tests/ensurePlaywrightBrowsers.test.ts:283:27
    281|       HTTPS_PROXY: envWithProxy.HTTPS_PROXY,
    282|     });
    283|     expect(writeFileSync).toHaveBeenCalledTimes(1);
       |                           ^
    284|   });
    285| 

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/4]⎯

 Test Files  2 failed | 289 passed (291)
      Tests  4 failed | 1035 passed (1039)
   Start at  18:01:05
   Duration  166.23s (transform 10.45s, setup 11.33s, collect 47.68s, tests 123.45s, environment 194.64s, prepare 37.25s)


Error: AssertionError: expected "spy" to be called with arguments: [ …(2) ]

Number of calls: 0

 ❯ frontend/tests/ensure-playwright-browsers.test.ts:94:35



Error: AssertionError: expected "spy" to be called at least once
 ❯ frontend/tests/ensure-playwright-browsers.test.ts:285:35



Error: AssertionError: expected "spy" to be called 1 times, but got 0 times
 ❯ scripts/tests/ensurePlaywrightBrowsers.test.ts:234:29



Error: AssertionError: expected "spy" to be called 1 times, but got 0 times
 ❯ scripts/tests/ensurePlaywrightBrowsers.test.ts:283:27


 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

new codex task, not a r/e/v/i/e/w task

--- a/frontend/scripts/utils/ensure-playwright-browsers.js
+++ b/frontend/scripts/utils/ensure-playwright-browsers.js
@@ -295,6 +295,13 @@ export function ensurePlaywrightSystemDeps(options = {}) {
 
     try {
         fsMkdirSync(path.dirname(depsStampPath), { recursive: true });
+    } catch (error) {
+        console.warn(
+            `Unable to create Playwright deps sentinel directory for ${depsStampPath}: ${error.message}`
+        );
+    }
+
+    try {
         fsWriteFileSync(depsStampPath, `${new Date().toISOString()}\n`);
     } catch (error) {
         console.warn(

--- a/frontend/src/components/__tests__/BuySell.spec.ts
+++ b/frontend/src/components/__tests__/BuySell.spec.ts
@@ -1,0 +1,84 @@
+import { render } from '@testing-library/svelte';
+import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
+
+const { dbGetMock } = vi.hoisted(() => ({
+    dbGetMock: vi.fn(),
+}));
+let consoleErrorSpy;
+
+vi.mock('../../utils/customcontent.js', () => ({
+    ENTITY_TYPES: { ITEM: 'item' },
+    db: { get: dbGetMock },
+}));
+
+vi.mock('../../pages/inventory/json/items', () => ({
+    default: [
+        { id: 'dusd-id', name: 'dUSD', price: '1 dUSD' },
+        { id: 'dcarbon-id', name: 'dCarbon', price: '1 dUSD' },
+    ],
+}));
+
+vi.mock('../../utils', () => ({
+    getPriceStringComponents: (price: string) => {
+        if (!price) {
+            return { price: 0, symbol: '' };
+        }
+
+        const [amount, symbol] = price.split(' ');
+        return { price: Number(amount), symbol: symbol ?? '' };
+    },
+}));
+
+vi.mock('../../utils/gameState/inventory.js', () => ({
+    buyItems: vi.fn(),
+    sellItems: vi.fn(),
+    getSalesTaxPercentage: vi.fn(() => 0),
+}));
+
+vi.mock('../../utils/gameState/common.js', () => ({
+    getPersistedGameStateLightweight: vi.fn(async () => ({ checksum: '' })),
+    syncGameStateFromLocalIfStale: vi.fn(),
+}));
+
+import BuySell from '../svelte/BuySell.svelte';
+
+describe('BuySell expected IndexedDB errors', () => {
+    beforeEach(() => {
+        dbGetMock.mockReset();
+        consoleErrorSpy = vi.spyOn(console, 'error').mockImplementation(() => {});
+    });
+
+    test('logs unexpected IndexedDB errors', async () => {
+        dbGetMock.mockRejectedValueOnce(new Error('IndexedDB error'));
+
+        const { findByText } = render(BuySell, {
+            itemId: 'custom-item',
+        });
+
+        await findByText('Item not found');
+
+        expect(consoleErrorSpy).toHaveBeenCalledWith(
+            'Failed to load item from IndexedDB in BuySell.svelte',
+            expect.objectContaining({
+                itemId: 'custom-item',
+                entityType: 'item',
+            })
+        );
+    });
+
+    test('suppresses known missing-item IndexedDB errors', async () => {
+        dbGetMock.mockRejectedValueOnce(new Error('item not found with id: custom-item'));
+
+        const { findByText } = render(BuySell, {
+            itemId: 'custom-item',
+        });
+
+        await findByText('Item not found');
+
+        expect(consoleErrorSpy).not.toHaveBeenCalled();
+    });
+
+    afterEach(() => {
+        consoleErrorSpy.mockRestore();
+    });
+});

